.github/workflows/build_docker_image.yml

```diff
@@ -1,64 +1,81 @@
     timeout-minutes: 20
     strategy:
       matrix:
-        image-name: [nxdk, nxdk-debug, nxdk-lto]
+        tag: [latest, debug, lto]
         include:
-          - image-name: nxdk
+          - tag: latest
             build-args: buildparams=CFLAGS=-O2 CXXFLAGS=-O2
-          - image-name: nxdk-debug
+          - tag: debug
             build-args: buildparams=DEBUG=y
-          - image-name: nxdk-lto
+          - tag: lto
             build-args: buildparams=LTO=y CFLAGS=-O2 CXXFLAGS=-O2
     steps:
       - name: Clone Tree
         uses: actions/checkout@v3
         with:
           submodules: recursive
-      - name: Generate timestamp
-        id: timestamp
-        run: echo ::set-output name=timestamp::$(date --rfc-3339=seconds --utc)
-      - name: Generate tags
-        id: tags
+      - name: Build Docker image
         run: |
-          DOCKER_IMAGE=xboxdev/${{ matrix.image-name }}
-          TAGS="${DOCKER_IMAGE}:latest"
-          TAGS="$TAGS,${DOCKER_IMAGE}:git-${GITHUB_SHA::8}"
-          TAGS="$TAGS,ghcr.io/${DOCKER_IMAGE}:latest"
-          TAGS="$TAGS,ghcr.io/${DOCKER_IMAGE}:git-${GITHUB_SHA::8}"
-          echo ::set-output name=tags::${TAGS}
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2
-      - name: Login to DockerHub
-        uses: docker/login-action@v2
-        with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
-      - name: Login to GitHub Container Registry
-        uses: docker/login-action@v2
-        with:
-          registry: ghcr.io
-          username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
-      - name: Build and push
-        uses: docker/build-push-action@v3
-        with:
-          context: .
-          file: ./Dockerfile
-          build-args: ${{ matrix.build-args }}
-          platforms: linux/amd64,linux/386
-          push: true
-          tags: ${{ steps.tags.outputs.tags }}
-          labels: |
-            org.opencontainers.image.revision=${{ github.sha }}
-            org.opencontainers.image.created=${{ steps.timestamp.outputs.timestamp }}
+          docker build --build-arg buildparams="${{ matrix.build-args }}" -t ${{ matrix.tag }} . \
+            --label "org.opencontainers.image.revision=${{ github.sha }}" \
+            --label "org.opencontainers.image.created=$(date --rfc-3339=seconds --utc)"
+      - name: Test Docker image
+        run: |
+          cd samples
+          for dir in */
+          do
+            cd "$dir"
+            docker run --rm -v `pwd`:/usr/src/app -t ${{ matrix.tag }} make -j`nproc`
+            cd ..
+          done
+      - name: Push container if on master branch
+        if: github.ref == 'refs/heads/master'
+        run: |
+          docker login -u "${{ github.actor }}" -p "${{ secrets.GITHUB_TOKEN }}" ghcr.io
+          docker tag ${{ matrix.tag }} ghcr.io/${{ github.repository }}:${{ matrix.tag }}
```

Comment:
This line is affected mentioned by https://github.com/XboxDev/nxdk/pull/656/files#r1278720076.
> Other flaw I see from this [PR changes is it push nxdk, the image, tag with (via latest, debug, lto)](https://github.com/sharkwouter/nxdk/pkgs/container/nxdk) as a commit tag. In order word, it messes with nxdk package as placeholder storage for the next matrix to able use it. And it will affect if multiple pushed/merged simultaneously which each will have different delay to start the process. I'm fairly sure this isn't the way we want it to do. Plus cause more confusion with nxdk's "nxdk" package's tags vs other packages which are currently in private I suspect. Using GH's artifact, cache, or other alternative are recommended to do this if we want to handle the docker image within GH's Action temporary before push to package registery as permanent storage.

Reply:
I made this a bit cleaner now. Hope it helps.